### ISSUE TITLE
Tweak Red Beams ("Shield Interference Beam", AKA "EMP beams")

### ISSUE THREAD

**Issue by Provolonage** [captaincobbs/ESG-Mod#1170]
They invalidate most of the combat sandbox once unlocked since they effectively give all beams 100% crit chance and everything gets 1 shot.

**Comment by Eighth-of-Third** [comment]
While we're at it, we should nerf Antimatter Lock-on. Stacking it with coordinator-only fleets is extremely strong in a similarly combat-breaking way. Maybe halve the damage boost or make it a unique boost per flotilla, if possible.

**Comment by FluffiestPrincess** [comment]
I believe - based on conversations in the Discord where this first came up - that Provolonage is talking about the module called "Shield Interference Beam" in flavour text and `ModuleWeaponBeamEMP1Strategic3` in the XML.

```xml
    <!-- EMP Weapon BEAM -->
  <SimulationDescriptor Name="ModuleWeaponBeamEMP1Strategic3" Type="ModuleWeapon">
    <Modifier TargetProperty="EMPDurationForShields"    Operation="Addition"    Value="6"         Path="ClassModule"       TooltipHidden="true"/>
    <Modifier TargetProperty="Damage"                   Operation="Addition"    Value="40"        Path="ClassModule"       TooltipHidden="true"/>
    <Modifier TargetProperty="Cooldown"                 Operation="Percent"     Value="0.5"       Path="ClassModule"       TooltipHidden="true"/>
    <BinaryModifier TargetProperty="TotalIndustryCost"    Operation="Addition" Left="$(WeaponIndustryTier3)" BinaryOperation="Multiplication" Right="$(Multiplier)" Path="ClassModuleWeapon" TooltipHidden="true"/>
    <BinaryModifier TargetProperty="TotalStrategic3Cost"    Operation="Addition" Left="1" BinaryOperation="Multiplication" Right="$(Multiplier)" Path="ClassModuleWeapon" TooltipHidden="true"/>

    <!-- Feedback -->
    <Modifier TargetProperty="EMPTargetShields"         Operation="Addition"    Value="1"           Path="ClassModule"      TooltipHidden="true"/>
  </SimulationDescriptor>
```

Since it's a beam weapon with 90% accuracy at all ranges, it does seem kind of difficult to balance. The EMP missiles can at least be shot down. And EMP is a binary state; we can't give it a chance to fail or require multiple hits to EMP a shield or make it only partly take down the shield.

Possible solutions:

- Make the Shield Interference Beam really really expensive or only able to mount on carrier hulls
- Scrap it and replace it with Shield Interference Missiles that can be shot down
- Scrap it and replace it with Shield Interference Slugs that have low accuracy at long range
- Make the Hardened Shield Generators module hardcounter shield EMPs
- Make the EMP duration really short and the cooldown very long and see if that helps

**Comment by Arpharnator** [comment]
no it's not this.
It's the red beam, the ones with 20% crit on the right side of the military tree stage 4.

**Comment by FluffiestPrincess** [comment]
That's "Advanced Positron Beam" (`ModuleWeaponBeam4Strategic4` for nerds), and I don't _think_ it's what Provolonage is talking about, based on the original Discord discussion where this came up and the EMP effect was mentioned.

**Comment by FluffiestPrincess** [comment]
Provolonage confirmed that they mean Shield Interference Beam, `ModuleWeaponBeamEMP1Strategic3`.

**Comment by FluffiestPrincess** [comment]
Of the options I suggested above, replacing this module with Shield Interference Missiles seems like the best option. As Eighth-of-Third pointed out, it is at least possible to guess that a carrier fleet with all beams but one random missile module is packing EMP, while the current "red beams" blend in with other beam weapons right up until they fire and you get instakilled.

It might still lead to degenerate strategies, but it's got to be worth a try.

(Meanwhile, Weapon Interference Missiles could actually become beams, and probably not be OP as long as you can't mount so many that you can completely lock down arbitrarily large fleets.)

An alternative suggestion that came up was replacing the total-shield-disabling effect with something else, which might or might not be possible depending on which part of the configuration is hardcoded.

**Comment by FluffiestPrincess** [comment]
One day, I will remember to change which branch I'm making the pull request to.

**Comment by FluffiestPrincess** [comment]
**Help wanted:** Please load the linked branch and check if the Shield Interference Missiles module actually triggers the shield disabling effect or not.

**Comment by FluffiestPrincess** [comment]
If people want, I can take a shot at coding an alternative version where the SIBeam is still a beam, but has a lesser effect than "totally disables shield".

**Comment by NoProblemWithThisAcc** [comment]
I could have sworn there was EMP protection, although I can't find them on the wiki. Why not just lower the damage on them?

Is there a way to make them also buff the armor's effect on lasers on the ships your are shooting? Thus low shield, high armor fleets will do well against EMPs when they would otherwise not, making EMP lasers more risky

**Comment by FluffiestPrincess** [comment]
> I could have sworn there was EMP protection, although I can't find them on the wiki. 

There is, it's called "Hardened Shield Generators", you can look up the effects under `ModuleSupportEMPShieldDefense1Strategic3`. It reduces the duration of shield EMP effects by 20%. This is not enough to make a difference.

> Why not just lower the damage on them?

The issue isn't damage. Shield Interference Beams cause an EMP effect on hit that entirely disables the target's shields. A subsequent hit from all of the _other_ energy weapons in the attacking fleet will generally shred the hull and destroy the target in one or two hits. We'd have to lower damage and rebalance absorbance values across the entire game.

> Is there a way to make them also buff the armor's effect on lasers on the ships your are shooting? Thus low shield, high armor fleets will do well against EMPs when they would otherwise not, making EMP lasers more risky

No, we're not doing that.

**Comment by FluffiestPrincess** [comment]
Hey, can someone please download the linked branch and test that the shield interference missiles actually work? I can't, because I don't have the right DLC.

**Comment by captaincobbs** [comment]
Does this just need testing? What should I look for

**Comment by wakystuf** [comment]
Yes, I believe that's correct. I'd make sure they are showing up properly on the tech tree and that they have the shield EMP effect as expected during a combat (slightly tricky to test, which I suspect is why nobody has yet done it)
